onnx/defs/object_detection/defs.cc

```diff
@@ -0,0 +1,114 @@
+    RoIAlign,
+    10,
+    OpSchema()
+      .SetDoc(RoIAlign_ver1_doc)
+      .Attr(
+          "spatial_scale",
+          "Multiplicative spatial scale factor to translate ROI coordinates "
+          "from their input spatial scale to the scale used when pooling, "
+          "i.e., spatial scale of the input feature map X relative to the "
+          "input image. E.g.; default is 1.0f. ",
+          AttributeProto::FLOAT,
+          1.f)
+      .Attr(
+          "output_height",
+          "default 1; Pooled output Y's height.",
+          AttributeProto::INT,
+          static_cast<int64_t>(1))
+      .Attr(
+          "output_width",
+          "default 1; Pooled output Y's width.",
+          AttributeProto::INT,
+          static_cast<int64_t>(1))
+      .Attr(
+          "sampling_ratio",
+          "Number of sampling points in the interpolation grid used to compute "
+          "the output value of each pooled output bin. If > 0, then exactly "
+          "sampling_ratio x sampling_ratio grid points are used. If == 0, then "
+          "an adaptive number of grid points are used (computed as "
+          "ceil(roi_width / output_width), and likewise for height). Default is 0.",
+          AttributeProto::INT,
+          static_cast<int64_t>(0))
+      .Input(
+          0,
+          "X",
+          "Input data tensor from the previous operator; "
+          "4-D feature map of shape (N x C x H x W), "
+          "where N is the batch size, C is the number of channels, "
+          "and H and W are the height and the width of the data.",
+          "T1")
+      .Input(
+          1,
+          "rois",
+          "RoIs (Regions of Interest) to pool over; rois is "
+          "2-D input of shape (num_rois, 4) given as "
+          "[[x1, y1, x2, y2], ...]. "
+          "The RoIs' coordinates are in the coordinate system of the input image.",
+          "T1")
+      .Input(
+          2,
+          "batch_splits",
+          "Tensor of shape (num_images) with each element denoting the number "
+          "of RoIs in the corresponding image in the batch.",
+          "T2")
+      .Output(
+          0,
+          "Y",
+          "RoI pooled output, 4-D tensor of shape "
+          "(num_rois, C, output_height, output_width). The r-th batch element Y[r-1] "
+          "is a pooled feature map corresponding to the r-th RoI X[r-1].",
+          "T1")
+      .TypeConstraint(
+          "T1",
+          {"tensor(float16)", "tensor(float)", "tensor(double)"},
+          "Constrain types to float tensors.")
+      .TypeConstraint(
+          "T2",
+          {"tensor(int16)", "tensor(int32)", "tensor(int64)", "tensor(uint8)",
+           "tensor(uint16)", "tensor(uint32)", "tensor(uint64)"},
+          "Constrain types to int tensors.")
+        .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
+          if (!hasNInputShapes(ctx, 3)) {
+            return;
+          }
+          propagateElemTypeFromInputToOutput(ctx, 0, 0);
```

Comment:
Similar to https://github.com/onnx/onnx/blob/c05f2ae412daf8fd64136ca354b97ccf73e0ea6c/onnx/defs/nn/defs.cc#L683, consider more checks (e.g.: input_shape.dim_size() == 4, rois_shape.dim_size() == 2, etc).